tests/test_compression.py

```diff
@@ -36,6 +36,11 @@
 ]
 
 
+def test_compression__does_not_raise_error_with_horizontal_line():
+    flat_data = np.array([list(range(100)), [10 for _ in range(100)]], dtype=np.int32)
+    compress_filtered_gmr(flat_data)
```

Comment:
can you assert that it filters everything out except the very first and very last point? it's always better to have an assertion in a test rather than just 'does not raise error'.

Reply:
For some reason it removes all points except the first point and last 3 points

Reply:
hmm, I guess it's a good thing we're adding the assertion then.  do you have any idea why it's keeping 3 points at the end? I don't think that's desired behavior...

Reply:
you can probably change the name of the test also (to something other than "does not raise error")

Reply:
I can try to debug. Will I need to understand the `rsquared` algorithm first or is this problem likely unrelated to that?

Reply:
I hope not.  if it's a horizontal line, then there's no reason that it shouldn't always be returning 1....as far as I can tell

Reply:
`r_squared` is correctly being set to 1 every time so the problem must be something else